src/Client/Provider/GithubClient.php

```diff
@@ -22,7 +22,29 @@ class GithubClient extends OAuth2Client
      */
     public function fetchUserFromToken(AccessToken $accessToken)
     {
-        return parent::fetchUserFromToken($accessToken);
+        $resourceOwner = parent::fetchUserFromToken($accessToken);
+
+        if (null === $resourceOwner->getEmail()) {
+            $response = $this->getOAuth2Provider()->getHttpClient()->request(
+                'GET',
+                'https://api.github.com/user/emails',
+                [
+                    'headers' => [
+                        'Authorization' => "Bearer {$accessToken->getToken()}"
+                    ]
+                ]
+            );
+
+            foreach (json_decode($response->getBody()->getContents(), true) as $email) {
+                if (true === $email['primary']) {
+                    return new GithubResourceOwner(\array_merge($resourceOwner->toArray(), [
```

Comment:
I think we don't need this `\` in front of `array_merge()`

Reply:
It's ok either way. There's a TINY performance boost in some cases with flat functions - so it's kind of hipster right now to add the `\` :p

Reply:
Tiny performance is always got to take especially on large applications. :-)

BTW, it's just a `\` addition. :wink: